src/renderer/corporate/member/corporate-add-member-container.jsx

```diff
@@ -1,64 +1,187 @@
 import React, { PureComponent } from 'react';
+import _ from 'lodash';
 import { CorporateAddMember } from './corporate-add-member';
 import { connect } from 'react-redux';
-import { addressBookSelectors, addressBookOperations } from 'common/address-book';
 import { withStyles } from '@material-ui/core/styles';
 import { push } from 'connected-react-router';
+import { appSelectors } from 'common/app';
+import { identityAttributes } from 'common/identity/utils';
+import { identityOperations, identitySelectors } from 'common/identity';
 
-const styles = theme => ({
-	radioBoxContainer: {
-		margin: '10px 0 40px'
-	}
-});
+const styles = theme => ({});
+
+const commonFields = ['shares', 'positions', 'type', 'did'];
+const corporateFields = [
+	'jurisdiction',
+	'taxId',
+	'entityType',
+	'email',
+	'entityName',
+	'creationDate'
+];
+const individualFields = [
+	'firstName',
+	'lastName',
+	'nationality',
+	'country',
+	'email',
+	'phoneNumber'
+];
 
 class CorporateAddMemberContainer extends PureComponent {
 	state = {
-		label: '',
-		address: ''
+		errors: { hasErrors: false },
+		type: 'individual'
 	};
 
-	componentDidMount() {
-		this.props.dispatch(addressBookOperations.resetAdd());
-	}
+	selectFields = type =>
+		type === 'individual'
+			? [...commonFields, ...individualFields]
+			: [...commonFields, ...corporateFields];
 
-	handleSubmit = event => {
-		event.preventDefault();
-		return this.handleSave(this.state.label, this.state.address);
-	};
+	handleFieldChange = name => evt => {
+		let value = evt;
 
-	handleSave = async (label, address) => {
-		await this.props.dispatch(addressBookOperations.addAddressBookEntry({ label, address }));
-		this.closeAction();
-	};
+		if (evt && evt.target) {
+			value = evt.target.value;
+		}
+
+		const stateErrors = { ...this.state.errors };
+		delete stateErrors[name];
+		const errors = this.validateAllAttributes([{ name, value }]);
 
-	handleLabelChange = event => {
-		event.preventDefault();
-		const label = event.target.value;
 		this.setState({
-			...this.state,
-			label
+			[name]: value
 		});
-		this.props.dispatch(addressBookOperations.validateLabel(label));
+		this.setErrors({ ...stateErrors, ...errors });
 	};
 
-	handleAddressChange = event => {
-		event.preventDefault();
-		const address = event.target.value;
+	setErrors(errors) {
+		const hasErrors = Object.keys(errors).length > 1;
 		this.setState({
-			...this.state,
-			address
+			errors: { ...errors, hasErrors }
 		});
-		this.props.dispatch(addressBookOperations.validateAddress(address));
+	}
+
+	validateAttributeType = type => {
+		return ['individual', 'corporate'].includes(type);
 	};
 
-	closeAction = () => {
-		this.props.dispatch(push('/main/create-corporate-profile'));
+	validateAttributePositions = selectedPositions => {
+		const acceptablePositions = this.props.positions.map(p => p.position);
```

Comment:
Instead of validating position, selected positions should be filtered by possible positions on change and before submit